2016/notes-on-testing-with-jest.md

```diff
@@ -0,0 +1,280 @@
+While we still have reservations of, if its actually feasible to review large changes in the snapshot, I definitely think this enriches the content of a code review.
+
+### Implementation Questions
+
+__How do I test my React+Redux code?__
+
+We have been following an approach where we test each of the moving parts independently.
+
+Components
+----------
+A React component is a view only component, with some event handling and some local state. It is `connect`ed to the Redux store for state and actions.
+We test the base component, which is free of Redux. For a component like,
+
+__my-component.js__
+
+```javascript
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+
+export class MyComponent extends Component {
+    constructor () {
+    super();
+    this.state = {
+        localProp: 'Default'
+    };
+    }
+    handleClick = () => {
+    this.setState({
+        localProp: 'After click'
+    });
+    }
+    render () {
+    return (<div>
+        <div>{this.props.reduxProp}</div>
+        <div>{this.state.localProp}</div>
+        <div onClick={this.handleClick}>Click me</div>
+    </div>);
+    }
+}
+
+export default connect(reduxState => ({reduxProp: reduxState.prop}) )(MyComponent)
+```
+
+__my-component.test.js__
+
+```javascript
+// needed for jsx
+import React from 'react';
+// import the named export
+import {MyComponent} from './my-component';
+// enzyme has rendering utils
+import {shallow} from 'enzyme';
+// needed to create snapshot
+import {shallowToJson} from 'enzyme-to-json';
+
+describe('<MyComponent />', () => {
+    let component;
+    // create an instance of MyComponent before each test
+    beforeEach(() => {
+    component = shallow(<MyComponent reduxProp="A test prop" />);
+    });
+    // test rendering expectations
+    it('should render well', () => {
+    // DOM test
+    expect(component.contains(<div>A test prop</div>)).toBe(true);
+    expect(component.contains(<div>Default</div>)).toBe(true);
+    // Match snapshots (creates the first time)
+    expect(shallowToJson(component)).toMatchSnapshot();
+    });
+    // test the action
+    it('should change text after click', () => {
+    //find particular DOM and simulate click
+    component.childAt(2).simulate('click');
+    test expectation after click is done
+    expect(component.text().indexOf('After click') > -1).toBe(true);
+    // match snapshot because view changed.
+    expect(shallowToJson(component)).toMatchSnapshot();
+    });
+});
```

Comment:
the indentation in this example seems off. I'd recommend to actually just drop the "describe" block too :D It's just overhead.